src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/bolts/OperationQueueBolt.java

```diff
@@ -0,0 +1,111 @@
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.reroute.bolts;
+
+import static org.openkilda.wfm.topology.utils.KafkaRecordTranslator.FIELD_ID_KEY;
+import static org.openkilda.wfm.topology.utils.KafkaRecordTranslator.FIELD_ID_PAYLOAD;
+
+import org.openkilda.messaging.MessageData;
+import org.openkilda.messaging.command.CommandData;
+import org.openkilda.messaging.command.CommandMessage;
+import org.openkilda.messaging.command.flow.FlowPathSwapRequest;
+import org.openkilda.messaging.command.flow.FlowRerouteRequest;
+import org.openkilda.messaging.info.reroute.PathSwapResult;
+import org.openkilda.messaging.info.reroute.RerouteResultInfoData;
+import org.openkilda.wfm.CommandContext;
+import org.openkilda.wfm.error.PipelineException;
+import org.openkilda.wfm.share.hubandspoke.CoordinatedBolt;
+import org.openkilda.wfm.topology.reroute.service.OperationQueueService;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+
+@Slf4j
+public class OperationQueueBolt extends CoordinatedBolt implements OperationQueueCarrier {
+    public static final String BOLT_ID = "operation-queue";
+
+    public static final String FLOW_ID_FIELD = RerouteBolt.FLOW_ID_FIELD;
+
+    public static final String REROUTE_QUEUE_STREAM = "reroute-queue-stream";
+    public static final Fields REROUTE_QUEUE_FIELDS = new Fields(FLOW_ID_FIELD, FIELD_ID_PAYLOAD, FIELD_ID_CONTEXT);
+
+    public static final Fields FLOW_HS_FIELDS = new Fields(FIELD_ID_KEY, FIELD_ID_PAYLOAD);
+
+    private transient OperationQueueService service;
+
+    public OperationQueueBolt(int defaultTimeout) {
+        super(true, defaultTimeout);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void init() {
+        this.service = new OperationQueueService(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void handleInput(Tuple tuple) throws PipelineException {
+        CommandContext context = pullContext(tuple);
+        MessageData data = pullValue(tuple, FIELD_ID_PAYLOAD, MessageData.class);
+        if (data instanceof FlowPathSwapRequest) {
+            FlowPathSwapRequest flowPathSwapRequest = (FlowPathSwapRequest) data;
+            service.addFirst(flowPathSwapRequest.getFlowId(), context.getCorrelationId(), flowPathSwapRequest);
+        } else if (data instanceof FlowRerouteRequest) {
+            FlowRerouteRequest flowRerouteRequest = (FlowRerouteRequest) data;
+            service.addLast(flowRerouteRequest.getFlowId(), context.getCorrelationId(), flowRerouteRequest);
+        } else if (data instanceof RerouteResultInfoData) {
+            RerouteResultInfoData rerouteResultInfoData = (RerouteResultInfoData) data;
+            service.operationCompleted(rerouteResultInfoData.getFlowId(), rerouteResultInfoData);
```

Comment:
Why you use 2 arguments if you read the first argument value from the second argument value?

PS Other service calls have the same issue with `flowId` arguments.

Reply:
We cannot get `flowId` from the second argument in the method itself, because the methods require `InfoData` or `CommandData`, which do not contain this field.